huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_domain_test.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+
+	"github.com/huaweicloud/golangsdk/openstack/waf_hw/v1/domains"
+)
+
+func TestAccWafDomainV1_basic(t *testing.T) {
+	var domain domains.Domain
+	resourceName := "huaweicloud_waf_domain.domain_1"
+	randName := acctest.RandString(8)
+	certificateName := acctest.RandString(8)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: testAccCheckWafDomainV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWafDomainV1_basic(certificateName, randName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafDomainV1Exists(resourceName, &domain),
+					resource.TestCheckResourceAttr(resourceName, "domain", fmt.Sprintf("www.%s.com", randName)),
+					resource.TestCheckResourceAttr(resourceName, "proxy", "false"),
+					resource.TestCheckResourceAttr(resourceName, "server.0.client_protocol", "HTTPS"),
+					resource.TestCheckResourceAttr(resourceName, "server.0.server_protocol", "HTTP"),
+					resource.TestCheckResourceAttr(resourceName, "server.0.port", "8080"),
+				),
+			},
+			{
+				Config: testAccWafDomainV1_update(certificateName, randName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafDomainV1Exists(resourceName, &domain),
+					resource.TestCheckResourceAttr(resourceName, "server.0.client_protocol", "HTTPS"),
+					resource.TestCheckResourceAttr(resourceName, "server.0.server_protocol", "HTTP"),
+					resource.TestCheckResourceAttr(resourceName, "server.0.port", "8443"),
+					//resource.TestCheckResourceAttr(resourceName, "proxy", "true"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"keep_policy"},
+			},
+		},
+	})
+}
+
+func TestAccWafDomainV1_policy(t *testing.T) {
+	var domain domains.Domain
+	resourceName := "huaweicloud_waf_domain.domain_1"
+	randName := acctest.RandString(8)
+	certificateName := acctest.RandString(8)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: testAccCheckWafDomainV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWafDomainV1_policy(certificateName, randName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafDomainV1Exists(resourceName, &domain),
+					resource.TestCheckResourceAttr(resourceName, "domain", fmt.Sprintf("www.%s.com", randName)),
+					resource.TestCheckResourceAttr(resourceName, "proxy", "true"),
+					resource.TestCheckResourceAttr(resourceName, "sip_header_name", "default"),
```

Comment:
sip_header_name is not defined in schema

Reply:
has deleted